src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -245,40 +246,83 @@ private static String getInstanceName(DetailAST methodCallAst) {
-                    && currentSiblingAst.getFirstChild().getType() == TokenTypes.METHOD_CALL) {
-                final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
-                final String instanceName = getInstanceName(methodCallAst);
-                if (instanceName.isEmpty()) {
-                    result = false;
-                }
-                else if (!instanceName.equals(initInstanceName)) {
-                    if (initInstanceName.isEmpty()) {
-                        initInstanceName = instanceName;
+            switch (currentSiblingAst.getType()) {
+                case TokenTypes.EXPR:
+                    if (currentSiblingAst.getFirstChild().getType() == TokenTypes.METHOD_CALL) {
+                        final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
+                        final String instanceName = getInstanceName(methodCallAst);
+                        if (initInstanceName.isEmpty()) {
+                            initInstanceName = instanceName;
+                        }
+                        isVarNotInInitializationSequence = isVarNotInInitializationSequence
+                            && !isVariableUsed(variableName, methodCallAst);
+                        result = areInstanceNamesEqual(initInstanceName, instanceName);
                     }
                     else {
                         result = false;
                     }
-                }
+                    break;
+                case TokenTypes.VARIABLE_DEF:
+                    final String currentVariableName =
+                            currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
+                    isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
+                    break;
 
+                default:
+                    result = currentSiblingAst.getType() == TokenTypes.SEMI;
+                    break;
             }
-            else if (currentSiblingAst.getType() == TokenTypes.VARIABLE_DEF) {
-                final String currentVariableName =
-                        currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
-                isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
+            currentSiblingAst = currentSiblingAst.getPreviousSibling();
+        }
+        boolean ans = result;
+        if (variableUsageAst != null) {
+            ans = result && !isVarNotInInitializationSequence;
+        }
+        return ans;
+    }
+
+    /**
+     * Checks if the provided instance names are equal.
+     *
+     * @param initInstanceName The init instance name.
+     * @param instanceName The instance name.
+     * @return true if the instance names is equal to initInstanceName.
+     */
+    private static boolean areInstanceNamesEqual(String initInstanceName, String instanceName) {
+        return !instanceName.isEmpty()
+                        && instanceName.equals(initInstanceName);
```

Comment:
What is the case where `initInstanceName` is empty?

Reply:
whenever in the sequence of the method first time that methodcall will be in method https://github.com/checkstyle/checkstyle/blob/3c6e287bb9d3c910e7173b0de5024ee94a0aabd4/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L249 at
```
                    if (currentSiblingAst.getFirstChild().getType() == TokenTypes.METHOD_CALL) {
                        final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
                        final String instanceName = getInstanceName(methodCallAst);
                        if (initInstanceName.isEmpty()) {
                            initInstanceName = instanceName;
                        }
```

in this case initInstanceName would be empty.

Reply:
I assume you are asking about `!instanceName.isEmpty()`

```
public class InstanceNameEmpty {
    public void testIssue32_6() {
        Option aOpt = null; // violation 'Distance between .* declaration and its first usage is 3.'
        Option bOpt = null; // violation 'Distance between .* declaration and its first usage is 2.'
        Option cOpt = null;
        isNull(cOpt); // distance = 1
        isNull(bOpt); // distance = 2
        isNull(aOpt); // distance = 3
    }

    class Option {

        public void setArgName(String string) {
        }

    }

    boolean isNull(Option o) {
        return false;}

}
```

for this type of case, `isNull` method is defined within the same class. Therefore, it's not an instance method but within the same class. Therefore, the instance name for isNull will be empty.

Reply:
It seems like I didn't ask my question well; let me rephrase: how can we have an "empty" name? Conceptually, this doesn't make sense.